File: viasoft.licensing.licensingmanagement-frontend/apps/licensing-management/src/app/pages/licensings/license-detail/notifications/licensed-tenant-saga-status-update-notification.ts
import { DataLivelyUpdate } from "@viasoft/common";
import { LicensingStatus } from "@viasoft/licensing-licensingmanagement-licensedtenant";
import { CurrentSagaStatus } from "@viasoft/licensing-management/app/tokens/enum/current-saga-status.enum";
import { LicensedTenantSagaInfo } from "@viasoft/licensing-management/app/tokens/interfaces/licensed-tenant-saga-info.interface";

export class LicensedTenantSagaStatusUpdateNotification implements DataLivelyUpdate {
    uniqueTypeName: string;
    amCreatingNewLicensedTenant: boolean;
    status: CurrentSagaStatus;
    currentLicensedTenantStatus: LicensingStatus;
    newEmail: string;
    currentTick: number;
}

File: viasoft.licensing.licensingmanagement/Viasoft.Licensing.LicensingManagement.Host/Sagas/AdministratorEmail/Data/LicensedTenantSagaStatusUpdateNotification.cs
using Viasoft.Licensing.LicensingManagement.Domain.DTOs.LicensedTenantSaga;
using Viasoft.Licensing.LicensingManagement.Domain.LicensedTenant.Enum;
using Viasoft.PushNotifications.Abstractions.Contracts;

namespace Viasoft.Licensing.LicensingManagement.Host.Sagas.AdministratorEmail.Data;

public class LicensedTenantSagaStatusUpdateNotification : NotificationUpdate
{
    public override string UniqueTypeName => "LicensedTenantSagaStatusUpdateNotification";
    public bool AmCreatingNewLicensedTenant { get; set; }
    public CurrentSagaStatus Status { get; set; }
    public LicensingStatus CurrentLicensedTenantStatus { get; set; }
    public string NewEmail { get; set; }
    public long CurrentTick { get; set; }
}